compiler/astutil/astutil.go

```diff
@@ -69,6 +76,12 @@ func FuncKey(d *ast.FuncDecl) string {
 	if star, ok := recv.(*ast.StarExpr); ok {
 		recv = star.X
 	}
+	if index, ok := recv.(*ast.IndexExpr); ok {
+		recv = index.X
+	}
+	if index, ok := recv.(*ast.IndexListExpr); ok {
+		recv = index.X
+	}
```

Comment:
Could a type switch be used here for brevity and clarity?

Reply:
Done.